cpp/src/arrow/compute/expression.h

```diff
@@ -56,6 +59,9 @@ class ARROW_EXPORT Expression {
     const Kernel* kernel = NULLPTR;
     std::shared_ptr<KernelState> kernel_state;
     TypeHolder type;
+    // Whether the entire call (including all its arguments) is selection-vector-aware
+    bool selection_vector_aware = false;
+    std::shared_ptr<SpecialForm> special_form = NULLPTR;
```

Comment:
Would `selection_vector_aware` defined in kernel?

Reply:
Yeah, a kernel will claim itself as `selection_vector_aware` here: https://github.com/apache/arrow/pull/42106/files#diff-c275e8c64a3935e84eaf970584e61b0d3db88fa5b052965505edf8126526c80eR527-R532
if it has special execution path to deal with the given selection vector directly.